### ISSUE TITLE
Building cvc4 modelBlocker on Windows

### ISSUE THREAD

**Issue by mudathirmahgoub** [CVC4/CVC4#2791]
When building the branch [https://github.com/ajreynol/CVC4/tree/modelBlocker](https://github.com/ajreynol/CVC4/tree/modelBlocker) for windows using the command 
`./configure.sh --win64 --static --gmp-dir=gmp-6.1.2 --no-unit-testing  --static-binary` the resulting binary is not working although `cvc4.exe -h` works. 

When the debug version is used 
`./configureCVC4 --win64 --static --gmp-dir=gmp-6.1.2 --no-unit-testing  --static-binary debug` 
the resulting binary works, but behaves differently than the Linux version  on the smt-lib code below. The Linux version returns 2 models while the Windows debug version fails an assertion. 
```
(set-logic ALL)
(set-option :produce-models true)
(set-option :incremental true)
(set-option :block-models true)
(set-option :finite-model-find true)
(set-option :sets-ext true)
(declare-sort Atom 0)
(declare-sort UnaryIntTup 0)
(declare-sort BinaryIntTup 0)
(declare-sort TernaryIntTup 0)
(declare-fun value_of_unaryIntTup (UnaryIntTup) (Tuple Int))
(declare-fun value_of_binaryIntTup (BinaryIntTup) (Tuple Int Int))
(declare-fun value_of_ternaryIntTup (TernaryIntTup) (Tuple Int Int Int))
(declare-fun atomNone () (Set (Tuple Atom)))
(declare-fun atomUniv () (Set (Tuple Atom)))
(declare-fun atomIden () (Set (Tuple Atom Atom)))
(declare-fun this_A () (Set (Tuple Atom)))
(declare-fun this_A0 () (Set (Tuple Atom)))
(declare-fun this_A1 () (Set (Tuple Atom)))
(assert (subset this_A0 this_A))
(assert (subset this_A1 this_A))
(assert (= (intersection this_A0 this_A1) atomNone))
(assert (= this_A (union this_A0 this_A1)))
(assert (= this_A atomUniv))
; fact$1
(assert (or (exists ((_x1 Atom)) (member (mkTuple _x1) this_A0)) (= this_A0 (as emptyset (Set (Tuple Atom))))))
; fact$2
(assert (or (exists ((_x2 Atom)) (member (mkTuple _x2) this_A1)) (= this_A1 (as emptyset (Set (Tuple Atom))))))
; Empty unary relation definition for Atom
(assert (= atomNone (as emptyset (Set (Tuple Atom)))))
; Universe definition for Atom
(assert (= atomUniv (as univset (Set (Tuple Atom)))))
; Identity relation definition for Atom
(assert (forall ((_a1 Atom)(_a2 Atom)) (=> (and (member (mkTuple _a1) atomUniv) (member (mkTuple _a2) atomUniv)) (= (member (mkTuple _a1 _a2) atomIden) (= _a1 _a2)))))

(check-sat)
(get-model)

(check-sat)
(get-model)
```

**Comment by mudathirmahgoub** [comment]
[cvc4 model blocker windows vs linux.zip](https://github.com/CVC4/CVC4/files/2756671/cvc4.model.blocker.windows.vs.linux.zip)

**Comment by 4tXJ7f** [comment]
From the video, it looks like there is an assertion failure in debug that causes the release version to just silently quit (it is not too surprising that there is this difference between debug and release versions as assertion checks are not done in release). I have tried to reproduce the issue but was not successful. If you can send me the binaries that you used, I can take another look (see Slack messages).

**Comment by 4tXJ7f** [comment]
After some investigation, we found out that the version that @mudathirmahgoub was using was not including the latest commit on the branch (b911c2648ab70f82e135c5b458abcdc3ea44e4f4) while the Linux version was. The earlier commit df5852e95164e20985758e87fd19415ef172002d results in the same assertion failure on Windows and Linux. As I pointed out above, the difference between debug and release versions is because the debug version is checking an assertion that the release version is lacking.
